File: memory-api/src/main/java/com/memory/dto/diary/response/DiaryResponse.java
package com.memory.dto.diary.response;

import com.memory.domain.diary.Diary;
import com.memory.dto.member.response.MemberResponse;
import lombok.Getter;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
public class DiaryResponse {
    private final Long id;
    private final String title;
    private final String content;
    private final LocalDate date;
    private final String mood;
    private final String weather;
    private final MemberResponse member;
    private final LocalDateTime createDate;
    private final LocalDateTime updateDate;

    private DiaryResponse(Long id, String title, String content, LocalDate date, 
                         String mood, String weather, MemberResponse member, 
                         LocalDateTime createDate, LocalDateTime updateDate) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.date = date;
        this.mood = mood;
        this.weather = weather;
        this.member = member;
        this.createDate = createDate;
        this.updateDate = updateDate;
    }

    public static DiaryResponse from(Diary diary) {
        if (diary == null) {
            return null;
        }

        return new DiaryResponse(
            diary.getId(),
            diary.getTitle(),
            diary.getContent(),
            diary.getDate(),
            diary.getMood(),
            diary.getWeather(),
            MemberResponse.from(diary.getMember()),
            diary.getCreateDate(),
            diary.getUpdateDate()
        );
    }
}

File: memory-api/src/main/kotlin/com/memory/dto/diary/response/DiaryResponse.kt
package com.memory.dto.diary.response

import com.memory.domain.diary.Diary
import com.memory.dto.member.response.MemberResponse
import java.time.LocalDate
import java.time.LocalDateTime

data class DiaryResponse private constructor(
    val id: Long?,
    val title: String?,
    val content: String?,      // 자바의 String은 null 가능성이 있어 ? 처리
    val date: LocalDate?,
    val mood: String?,
    val weather: String?,
    val member: MemberResponse?,
    val createDate: LocalDateTime?,
    val updateDate: LocalDateTime?,
) {
    companion object {
        @JvmStatic
        fun from(diary: Diary?): DiaryResponse? {
            if (diary == null) return null

            return DiaryResponse(
                id = diary.id,
                title = diary.title,
                content = diary.content,
                date = diary.date,
                mood = diary.mood,
                weather = diary.weather,
                member = diary.member?.let { MemberResponse.from(it) },
                createDate = diary.createDate,
                updateDate = diary.updateDate
            )
        }
    }
}